code/datums/components/atmospheric_scanner.dm

```diff
@@ -0,0 +1,58 @@
+///Atoms with this component can signal it in order to get a scan of surrounding atmospherics sent to a specific mob.
+/datum/component/atmospheric_scanner
+	/// Controls if the analyzer requires being able to see it in order to obtain the results. The value is set in AddComponent.
+	var/visible = TRUE
+
+/datum/component/atmospheric_scanner/Initialize(visible)
+	if(!isitem(parent))
+		return COMPONENT_INCOMPATIBLE
+	if(visible)
```

Comment:
This doesn't make any sense. It defaults to TRUE, but if it's TRUE, then set it to TRUE?

```suggestion
	if(!isnull(visible))
```

Reply:
right, null isnt the same as false, my bad